ex4/ex4.c

```diff
@@ -6,10 +6,26 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <string.h>
 
 int main(int argc, char* argv[])
 {
-    // Your code here    
+    int rc = fork();
+
+    if (rc < 0)
+    {
+        fprintf(stderr, "Fork has failed\n");
+        exit(1);
+    } else if (rc == 0) 
+    {
+        char *myArg = {"/bin/ls"};
```

Comment:
I pulled down this code and ran it. Looks like the `execv()` call doesn't go through. I made a few changes and got it to run. It looks like this now in the child process's if block:
```
char *myArg[] = {"/bin/ls", (char *) NULL};
printf("This is the child program\n");
execv("/bin/ls", myArg);
```
I had to change the `myArg` declaration to be an array in order to store a NULL character pointer at the end. `exec` needs this in order to know when to stop looking for additional parameters. Then as a result, I also had to change how the `myArg` array was passed into the `execv()` call. 

That got it actually executing the `ls` program.